diff --git a/src/EventSubscriber/ExceptionSubscriberBase.php b/src/EventSubscriber/ExceptionSubscriberBase.php
index 9054b03..0b00f97 100644
--- a/src/EventSubscriber/ExceptionSubscriberBase.php
+++ b/src/EventSubscriber/ExceptionSubscriberBase.php
@@ -95,12 +95,12 @@ abstract class ExceptionSubscriberBase implements EventSubscriberInterface {
   /**
    * Handles exceptions for this subscriber.
    *
-   * @param \Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent $event
+   * @param \Symfony\Component\HttpKernel\Event\ExceptionEvent $event
    *   The event to process.
    */
-  public function onException(GetResponseForExceptionEvent $event) {
+  public function onException(\Symfony\Component\HttpKernel\Event\ExceptionEvent $event) {
     if ($this->applies($event->getRequest())) {
-      $exception = $event->getException();
+      $exception = $event->getThrowable();
       $response = $this->exceptionToResponse($exception);
       $event->setResponse($response);
     }
diff --git a/tests/src/Unit/DefaultExceptionSubscriberTest.php b/tests/src/Unit/DefaultExceptionSubscriberTest.php
index 441b3cd..7296b80 100644
--- a/tests/src/Unit/DefaultExceptionSubscriberTest.php
+++ b/tests/src/Unit/DefaultExceptionSubscriberTest.php
@@ -16,6 +16,7 @@ use Symfony\Component\HttpKernel\HttpKernelInterface;
  */
 class DefaultExceptionSubscriberTest extends UnitTestCase {
 
+  use \Prophecy\PhpUnit\ProphecyTrait;
   /**
    * The mocked HTTP kernel.
    *
@@ -26,7 +27,7 @@ class DefaultExceptionSubscriberTest extends UnitTestCase {
   /**
    * {@inheritdoc}
    */
-  public function setUp() {
+  public function setUp(): void {
     parent::setUp();
 
     $this->kernel = $this->prophesize(HttpKernelInterface::class)->reveal();
@@ -39,7 +40,7 @@ class DefaultExceptionSubscriberTest extends UnitTestCase {
     $request = Request::create('/whoops');
 
     $exception = new \RuntimeException('Houston, we have a problem');
-    $event = new GetResponseForExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
+    $event = new \Symfony\Component\HttpKernel\Event\ExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
 
     $subscriber = new DefaultExceptionSubscriber();
     $subscriber->onException($event);
@@ -61,7 +62,7 @@ class DefaultExceptionSubscriberTest extends UnitTestCase {
     $request->setRequestFormat('unknown');
 
     $exception = new \Exception('Houston, we have a problem');
-    $event = new GetResponseForExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
+    $event = new \Symfony\Component\HttpKernel\Event\ExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
 
     $subscriber = new DefaultExceptionSubscriber();
     $subscriber->onException($event);
@@ -87,7 +88,7 @@ class DefaultExceptionSubscriberTest extends UnitTestCase {
       'X-Custom' => 'test',
     ];
     $exception = new HttpException(406, 'Houston, we have a problem', NULL, $headers);
-    $event = new GetResponseForExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
+    $event = new \Symfony\Component\HttpKernel\Event\ExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
 
     $subscriber = new DefaultExceptionSubscriber();
     $subscriber->onException($event);
diff --git a/tests/src/Unit/JsonExceptionSubscriberTest.php b/tests/src/Unit/JsonExceptionSubscriberTest.php
index ccd8320..6598fcb 100644
--- a/tests/src/Unit/JsonExceptionSubscriberTest.php
+++ b/tests/src/Unit/JsonExceptionSubscriberTest.php
@@ -16,6 +16,7 @@ use Symfony\Component\HttpKernel\HttpKernelInterface;
  */
 class JsonExceptionSubscriberTest extends UnitTestCase {
 
+  use \Prophecy\PhpUnit\ProphecyTrait;
   /**
    * The mocked HTTP kernel.
    *
@@ -26,7 +27,7 @@ class JsonExceptionSubscriberTest extends UnitTestCase {
   /**
    * {@inheritdoc}
    */
-  public function setUp() {
+  public function setUp(): void {
     parent::setUp();
 
     $this->kernel = $this->prophesize(HttpKernelInterface::class)->reveal();
@@ -40,7 +41,7 @@ class JsonExceptionSubscriberTest extends UnitTestCase {
     $request->setRequestFormat('json');
 
     $exception = new \RuntimeException('Houston, we have a problem');
-    $event = new GetResponseForExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
+    $event = new \Symfony\Component\HttpKernel\Event\ExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
 
     $subscriber = new JsonExceptionSubscriber();
     $subscriber->onException($event);
@@ -68,7 +69,7 @@ class JsonExceptionSubscriberTest extends UnitTestCase {
     $request->setRequestFormat('hal_json');
 
     $exception = new \RuntimeException('Houston, we have a problem');
-    $event = new GetResponseForExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
+    $event = new \Symfony\Component\HttpKernel\Event\ExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
 
     $subscriber = new JsonExceptionSubscriber();
     $subscriber->onException($event);
@@ -96,7 +97,7 @@ class JsonExceptionSubscriberTest extends UnitTestCase {
     $request->setRequestFormat('html');
 
     $exception = new \Exception('Houston, we have a problem');
-    $event = new GetResponseForExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
+    $event = new \Symfony\Component\HttpKernel\Event\ExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
 
     $subscriber = new JsonExceptionSubscriber();
     $subscriber->onException($event);
@@ -116,7 +117,7 @@ class JsonExceptionSubscriberTest extends UnitTestCase {
       'X-Custom' => 'test',
     ];
     $exception = new HttpException(406, 'Houston, we have a problem', NULL, $headers);
-    $event = new GetResponseForExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
+    $event = new \Symfony\Component\HttpKernel\Event\ExceptionEvent($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $exception);
 
     $subscriber = new JsonExceptionSubscriber();
     $subscriber->onException($event);
